packages/toolpad-app/src/components/AppEditor/PageEditor/RenderPanel.tsx

```diff
@@ -688,7 +688,8 @@ export default function RenderPanel({ className }: RenderPanelProps) {
 
           const isOnlyRowElement =
             dragParent &&
-            getElementNodeComponentId(dragParent as appDom.ElementNode) === PAGE_ROW_COMPONENT_ID &&
+            (dragParent as appDom.ElementNode).attributes.component?.value ===
```

Comment:
is there a better way to get around the types here?
the default type doesn't let me try to use `component`

Reply:
The parent of an element can either be a `PageNode` or another `ElementNode`. A `PageNode` doesn't have a `component` attribute. That's why typescript is complaining here.

There are methods like `appDom.isPage`, or `appDom.isElement` that can act as type guards. You can use it in a  conditional to get rid of the type warnings without resorting to an `as` type assertion.

Reply:
ok, i'll keep the `as` if it's ok, so i don't to have to nest part of the logic inside an `if` - i feel like the way i wrote it is pretty explicit but lmk if you disagree

Reply:
ok i'll add the checks then, i get your point too